pkg/platform/kube/ingress/ingress.go

```diff
@@ -0,0 +1,423 @@
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/nuclio/nuclio/pkg/cmdrunner"
+	"github.com/nuclio/nuclio/pkg/common"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+)
+
+// keeps resources needed for ingress creation
+// (secret is used when it is an ingress with basic-auth authentication)
+type Resources struct {
+	Ingress *v1beta1.Ingress
+	Secret *v1.Secret
+}
+
+type Manager struct {
+	logger        logger.Logger
+	cmdRunner     cmdrunner.CmdRunner
+	kubeClientSet kubernetes.Interface
+}
+
+func NewManager (parentLogger logger.Logger,
+	kubecClientSet kubernetes.Interface,
+	cmdRunner cmdrunner.CmdRunner) (*Manager, error) {
+
+	return &Manager{
+		logger:        parentLogger.GetChild("runner"),
+		cmdRunner:     cmdRunner,
+		kubeClientSet: kubecClientSet,
+	}, nil
+}
+
+func (im *Manager) GenerateIngressResources(ctx context.Context,
+	spec Spec) (*Resources,  error) {
+
+	var err error
+	var secretResource *v1.Secret
+
+	ingressAnnotations := map[string]string{
+		"kubernetes.io/ingress.class": "nginx",
+	}
+	if spec.RewriteTarget != "" {
+		ingressAnnotations["nginx.ingress.kubernetes.io/rewrite-target"] = spec.RewriteTarget
+	}
+
+	if spec.UpstreamVhost != "" {
+		ingressAnnotations["nginx.ingress.kubernetes.io/upstream-vhost"] = spec.UpstreamVhost
+	}
+
+	if spec.BackendProtocol != "" {
+		ingressAnnotations["nginx.ingress.kubernetes.io/backend-protocol"] = spec.BackendProtocol
+	}
+
+	if spec.SSLPassthrough {
+		ingressAnnotations["nginx.ingress.kubernetes.io/ssl-passthrough"] = "true"
+	} else {
+		var authIngressAnnotations map[string]string
```

Comment:
can that move onto `compileAuthAnnotations` to make this function shorter and readable?